src/index.spec.ts

```diff
@@ -382,6 +394,167 @@ describe('EppoJSClient E2E test', () => {
+      const isolatedClient = new EppoJSClient(options);
+
+      expect(isolatedClient).not.toEqual(getInstance());
+      await isolatedClient.waitForReady();
+
+      expect(isolatedClient.isInitialized()).toBe(true);
+      expect(isolatedClient.initialized).toBe(true);
+      expect(getInstance().isInitialized()).toBe(false);
+      expect(getInstance().initialized).toBe(false);
+
+      expect(getInstance().getStringAssignment(flagKey, 'subject-10', {}, 'default-value')).toEqual(
+        'default-value',
+      );
+      expect(
+        isolatedClient.getStringAssignment(flagKey, 'subject-10', {}, 'default-value'),
+      ).toEqual('variant-1');
+    });
+    it('initializes on instantiation and notifies when ready', async () => {
+      const apiOptions: IApiOptions = { sdkKey: '<MY SDK KEY>', baseUrl };
+      const options: IClientOptions = { ...apiOptions, assignmentLogger: mockLogger };
+      const client = new EppoJSClient(options);
+
+      expect(client.getStringAssignment(flagKey, 'subject-10', {}, 'default-value')).toEqual(
+        'default-value',
+      );
+
+      await client.waitForReady();
+
+      const assignment = client.getStringAssignment(flagKey, 'subject-10', {}, 'default-value');
+      expect(assignment).toEqual('variant-1');
+    });
+  });
+
+  describe('multiple client instances', () => {
+    const API_KEY_1 = 'my-api-key-1';
+    const API_KEY_2 = 'my-api-key-2';
+    const API_KEY_3 = 'my-api-key-3';
+
+    const commonOptions: Omit<IClientOptions, 'sdkKey'> = {
+      baseUrl,
+      assignmentLogger: mockLogger,
+    };
+
+    let callCount = 0;
+
+    beforeAll(() => {
+      global.fetch = jest.fn((url: string) => {
+        callCount++;
+
+        const urlParams = new URLSearchParams(url.split('?')[1]);
+
+        // Get the value of the apiKey parameter and serve a specific variant.
+        const apiKey = urlParams.get('apiKey');
+
+        // differentiate between the SDK keys by changing the variant that `flagKey` assigns.
+        let variant = 'variant-1';
+        if (apiKey === API_KEY_2) {
+          variant = 'variant-2';
+        } else if (apiKey === API_KEY_3) {
+          variant = 'variant-3';
+        }
+
+        const encodedVariant = base64Encode(variant);
+
+        // deep copy the mock data since we're going to inject a change below.
+        const flagConfig: Flag = JSON.parse(JSON.stringify(mockObfuscatedUfcFlagConfig));
+        // Inject the encoded variant as a single split for the flag's only allocation.
+        flagConfig.allocations[0].splits = [
+          {
+            variationKey: encodedVariant,
+            shards: [],
+          },
+        ];
+
+        const ufc = { flags: { [obfuscatedFlagKey]: flagConfig } };
+
+        return Promise.resolve({
+          ok: true,
+          status: 200,
+          json: () => Promise.resolve(ufc),
+        });
+      }) as jest.Mock;
+    });
+    afterAll(() => {
```

Comment:
minor/linter but consider adding a space between this function and the one above